app/Http/Controllers/GroupsController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Group;
+use App\Models\User;
+use App\Models\Groups_Users;
+use App\Http\Requests\StoreGroupRequest;
+use App\Http\Requests\DeleteGroupRequest;
+use App\Http\Requests\JoinGroupRequest;
+use App\Http\Requests\LeaveGroupRequest;
+use App\Http\Resources\GroupResource;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
+
+
+class GroupsController extends Controller
+{
+    public function show($argument){
+        if ($argument == 'all') {
+            $allGroups = Group::where('is_public', true)->get();
+            foreach ($allGroups as $group){
+                $group->members = $group->users->map->only('id', 'username', 'rank')->map(function ($member) use ($group) {
+                    $member['rank'] = $group->rankOfMember(User::find($member['id']));
+                    return $member;
+                });
+            }
+            return GroupResource::collection($allGroups);
+        }
+        if ($argument == 'my') {
+            $myGroups = Auth::user()->groups;
+            foreach ($myGroups as $group){
+                $group->members = $group->users->map->only('id', 'username', 'rank')->map(function ($member) use ($group) {
+                    $member['rank'] = $group->rankOfMember(User::find($member['id']));
```

Comment:
- [ ] The `rankOfMember` expects a User but only uses an id. Since you already have this id in `$member['id']`, you don't need to fetch the entire User

Reply:
method is called `rankOfMemberById`